### PR TITLE
feat: search  & category feature

### PR DESCRIPTION
**Pull request by PeeyushPrashant**
- filter videos upon clicking on categories
- filter videos according to searched keyword

Screenshot (165)
Screenshot (166)

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -35,17 +35,26 @@
 }
 .right-main {
   padding: 3rem 0rem 3rem 0;
+  /* display: grid;
```
Comment (vinc3nati) [inline src/App.css:38]: remove unwanted code

File: src/pages/SingleVideo/SingleVideo.jsx [hunk 2]
```diff
@@ -49,28 +49,28 @@ export const SingleVideo=()=>{
                     onClick={!token?()=>navigate("/login"):()=>LikedHandler(video,token,dispatch,isInLiked)}
```
Comment (vinc3nati) [inline src/pages/SingleVideo/SingleVideo.jsx:49]: try to put only one liners here, else put the code in a separate function

File: src/App.css (hunk 1 above)
Comment (vinc3nati) [inline src/App.css:38]: remove unwanted code

**Review by vinc3nati — APPROVED** [review]

**Review by thegeorgenikhil — APPROVED** [review]
Good to merge
- use netlify for deploy previews

File: src/reducer/DataReducer.js [hunk 3]
```diff
@@ -2,16 +2,39 @@ import { ACTION_TYPE } from "../utils/ActionType";
...
+          ...cat,
+          isActive: cat.categoryName === "All" ? true : false,
+        })),
+      };
+    case ACTION_TYPE.SORT_BY:
+      return {
+        ...state,
+        sortBy: action.payload,
+        categories: state.categories.map((cat) =>
+          cat.categoryName === action.payload
```
Comment (thegeorgenikhil) [inline src/reducer/DataReducer.js:31]: use category instead of cat for better variable naming

**Review by Rohanmond — APPROVED** [review]
you can go ahead

**Pull request merged**